src/main/java/com/codeborne/selenide/collections/ExactTexts.java

```diff
@@ -49,7 +50,11 @@ public void fail(WebElementsCollection collection, List<WebElement> elements, Ex
       ElementNotFound elementNotFound = new ElementNotFound(collection, expectedTexts, lastError);
       elementNotFound.timeoutMs = timeoutMs;
       throw elementNotFound;
-    } else {
+    }
+    if (elements.size() != expectedTexts.size()) {
```

Comment:
Better to use `if { } else if { } else { }` construction in this case. Also please leave condition verification in one line as it was before

Reply:
fixed